import "nanopb.proto";

message LogisticsInfo
{
    //GUID of the platform
    optional string platformGuid = 1 [(nanopb).max_size = 40];
    
    //GUID of the sensor
    optional string sensorGuid = 2 [(nanopb).max_size = 40];
}

message HeaderMessage
{
    required uint32 version = 1;
    
    enum MessageType{
        DISCOVER = 0;
        TEMPERATURE = 1;
    }
        
    required MessageType message_type = 2;
    
    //Where this message is destined for
    optional LogisticsInfo destination = 3;
    
    //Where this message came from
    optional LogisticsInfo origination = 4;
}